src/platform/selfsign_openssl.c

```diff
@@ -85,27 +86,27 @@ static uint8_t* ReadPkcs12(const char* Name, uint32_t* Length) {
 }
 
 //
-// Generates a self signed cert using low level OpenSSL APIs.
+// Generates a (self) signed cert using low level OpenSSL APIs.
 //
+
+
 QUIC_STATUS
-CxPlatTlsGenerateSelfSignedCert(
-    _In_z_ char *CertFileName,
-    _When_(OutputPkcs12 == TRUE, _Reserved_)
-    _When_(OutputPkcs12 == FALSE, _In_z_)
-        char *PrivateKeyFileName,
+CxPlatTlsGenerateSignedCert(
+    _In_opt_z_ X509 *CaX509,
```

Comment:
This is not a null-terminated byte buffer, so you'd use `_In_opt_` instead.

Reply:
Also, consider renaming this to `SignerCert` or similar to state the purpose more clearly.

Reply:
Sorry, I was guessing what the _In_*_ annotations meant. Should have looked it up.